dev_guide/builds.adoc

```diff
@@ -1426,6 +1432,94 @@ Configuration change triggers currently only work when creating a new
 able to launch a build whenever a `*BuildConfig*` is updated.
 ====
 
+[[build-hooks]]
+
+== Build Hooks
+
+Build hooks can be used as a part of the build flow to run tests inside a
+temporary container and validate your image before it is being pushed to a
+registry. The command is executed with the current working directory set to the
+image's `*WORKDIR*` which is your current working directory where your code is
+located. The hook will fail if the script or command returns a non-zero exit
+code or if there's any other error related to starting the temporary container.
+When the hook fails it will mark the build as failed.
+====
+[source,yaml]
+spec:
+  postCommit:
+    script: "rake test" <1>
+
+<1> `script` is a shell script to be run with `/bin/sh -c`. Use this when a
+shell script is appropriate to execute the post build hook, for example for
+running unit tests with "rake test". If you need control over the image
+entrypoint, or if the image does not have "/bin/sh", use `command` and/or
+`args`.
+
+[source,yaml]
+spec:
+  postCommit:
+    command: ["/bin/sh", "-c"] <2>
+    args: ["rake, ,"test"] <3>
+
+<2> `command` is the command to run. This might be needed if the image doesn't
+have "/bin/sh" or if you do not want to use a shell. In all other cases, using
+`script` might be more convenient.
```

Comment:
additional final sentence.  "This value is used as the entrypoint for the container."